rumqttc/src/eventloop.rs

```diff
@@ -312,6 +312,15 @@ pub(crate) async fn socket_connect(
             socket.set_recv_buffer_size(recv_buffer_size).unwrap();
         }
 
+        socket
+            .bind_device(
+                network_options
+                    .bind_device
+                    .as_ref()
+                    .map(|bind_device| bind_device.as_bytes()),
+            )
+            .unwrap();
```

Comment:
https://docs.rs/tokio/latest/tokio/net/struct.TcpSocket.html#method.bind_device this method is available for specific platforms only, so we should only include this if we are building for that target.

Reply:
Also added docs to the set function